### PR TITLE
Add wikipedia source prototype

### PR DESCRIPTION
**Pull request by enzoMagalhaes**
# Description

Created a Wikipedia source prototype that extracts data from articles on Wikipedia. The WikipediaSource class includes methods for searching Wikipedia, retrieving data for a single Wikipedia page, and fetching data from multiple related Wikipedia pages. The data is stored in a DataFrame with specified columns.

searching example:
image

fetching a single page example:
image

fetching multiple pages related to a search term example:
image

# Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

unit tests were made, they can be found in the tests/test_wikipedia.py file. The functionality has been tested using the following steps:

- Create an instance of the WikipediaSource class.
- Perform a Wikipedia search for a specific term and assert that the results are returned correctly.
- Retrieve data from a single Wikipedia page and assert that the correct data is inserted into the Dataframe.
- Fetch data from multiple related Wikipedia pages and assert that all Wikipedia pages data are fetched.

## Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

unit tests:
- [x] Execute command pytest

manual tests:
- [x] Execute wikipedia_example.py
- [x] Confirm search results are obtained.
- [x] Verify data is correctly fetched from both a single page and multiple related pages.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by olivierdupuis — CHANGES_REQUESTED** [review]
I changed the branch to merge into. I want us to merge into a `dev` branch, not a `release` branch as those are already released.

This is a great PR. PR template was completed perfectly. And agree with you, structuring around a class is more elegant. I'll eventually refactor GDELT to follow that convention.

I see that you haven't added unit tests. It's probably a good call. I didn't feel like it added a whole lot of benefits to my own dev. Just having a good enough example is good. 

One small thing to change. Can you add a link to your wikipedia_example on the main examples page: https://github.com/enzoMagalhaes/social_signals/blob/wikipedia_source/examples/index.md

Besides that, it looks good. Good job :) Once you make that small change, request another review from me and I'll approve.

Btw, when you open a PR, you need to add me as a reviewer for me to get notified. Also, once I approve, you will be the one that merge the branch into the dev branch. I want the branch author to be responsible for the merge at the end.

File: pyproject.toml [hunk 1]
```diff
@@ -7,14 +7,16 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.9"
-pandas = "^1.5.3"
+pandas = "2.1.3"
```
Comment (olivierdupuis) [inline pyproject.toml:10]: Can you change it back to 1.5.3. There are compatibility issues with version 2+.

Comment (enzoMagalhaes) [reply pyproject.toml]: Just downgraded the pandas version back to 1.5.3 :+1:

**Review by olivierdupuis — APPROVED** [review]
Looks good @enzoMagalhaes 👍  You can merge!

**Pull request merged**